### ISSUE TITLE
Lighting/shadowing issue in Borderlands 2 (may be NVidia only)

### ISSUE THREAD

**Issue by edwin-v** [Joshua-Ashton/d9vk#178]
Borderlands 2 shows several (probably related) visual artefacts.

First artefacts are patters in darker areas. Especially noticeable on the white surfaces in the snow area.  The patterns move around and change size depending on camera position. It looks like this:

![Patterns](https://steamuserimages-a.akamaihd.net/ugc/786355787333555684/3F55BF4826CF86C882F9666167BEAFF088521D23/)

Similar patterns appear at the edges of shadows.

A second artefact is darker bars at appears at a fixed distance (probably more a fixed Z coord) from the camera. The repeat in the distance, but only the closest one is clearly visible in a static screenshot:

![Bars](https://steamuserimages-a.akamaihd.net/ugc/784104461447420068/762FD3A132637EB52906CBD9328D18765743E61D/)

These are not that easy to see on screenshots, but very noticeable when moving.

### Software information
Borderlands 2 / Steam. Max settings at 3440x1440@100

### System information
- GPU: RTX 2080 Ti
- Driver: 430.26
- Wine version: Proton 4.2-7
- DXVK version: 0.11 - master

### Apitrace file(s)
- If this is indeed an nvidia issue (can others confirm this really doesn't occur on amd?) then apitraces are probably not that useful.

### Log files
- d3d11.log: [Launcher_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3303386/Launcher_d3d9.log)

- dxgi.log:

**Comment by edwin-v** [comment]
Just tested the frog cookie release that was included in Proton 4.11-3 and both issues are still unchanged.

Driver is now 430.40 though. Also tried different settings, but it is unaffected by any of them.

**Comment by dlshinobi** [comment]
> Just tested the frog cookie release that was included in Proton 4.11-3 and both issues are still unchanged.
> 
> Driver is now 430.40 though. Also tried different settings, but it is unaffected by any of them.

Found the solution to this problem, you need "d3d9.supportDFFormats = False" in dxvk.conf. This also fixes the issue in https://github.com/Joshua-Ashton/d9vk/issues/254

**Comment by edwin-v** [comment]
Not sure how to test this since it appears that D9VK won't load custom configs in Proton.

**Comment by dlshinobi** [comment]
> Not sure how to test this since it appears that D9VK won't load custom configs in Proton.

I just created dxvk.conf inside folder where Borderlands 2.exe exist.
For me it's: "../steamapps/common/Borderlands 2/Binaries/Win32/"

Because I did Borderland2.exe -> Launcher.exe, so my dxvk.conf content looks like this:

[Launcher.exe] 
d3d9.supportDFFormats = False

**Comment by edwin-v** [comment]
Interesting. That's the current directory for the game where I thought it had to go.

Anyway, the option does change something, but it doesn't fix it. After setting that option I get larger patterns in the shadow edges. It may actually be worse than before because with that option I see a clear jump in the pattern scale when you move closer or further away. Which does suggest that this problem has to do with shadow maps.

**Comment by dlshinobi** [comment]
> Interesting. That's the current directory for the game where I thought it had to go.
> 
> Anyway, the option does change something, but it doesn't fix it. After setting that option I get larger patterns in the shadow edges. It may actually be worse than before because with that option I see a clear jump in the pattern scale when you move closer or further away. Which does suggest that this problem has to do with shadow maps.

Really? For me it fixes everything: shadow, those ugly lines in front on you. Not a single graphical bug for me. It's in perfect condition.
![Screenshot_20190907_223122](https://user-images.githubusercontent.com/7222444/64479395-48b08f00-d1bf-11e9-96ae-dbb4d6eb0648.png)

**Comment by edwin-v** [comment]
Look at the ice block in the middle of the screen as I move away from it:

![20190907215638_1](https://user-images.githubusercontent.com/1945340/64479675-7a732700-d1ba-11e9-8cdb-e93f6abe3d60.jpg)
![20190907215641_1](https://user-images.githubusercontent.com/1945340/64479676-7e9f4480-d1ba-11e9-9c50-3b8b109320a5.jpg)
![20190907215648_1](https://user-images.githubusercontent.com/1945340/64479683-8363f880-d1ba-11e9-99ad-e63c37bff94e.jpg)

**Comment by dlshinobi** [comment]
You are right, I didn't notice this. I only played DLC's and they are dark and I missed it. Stupid, stupid eyes! But this approach fixes more then before. Also I notices not every shadows are shimmering.

**Comment by Joshua-Ashton** [comment]
Please re-test with: https://github.com/doitsujin/dxvk/commit/b738c4220b7f2ea71cf2f1f61521a08606ee7022

**Comment by edwin-v** [comment]
Unfortunately that made no difference.

Both the bar and the patterns are still the same.

To be honest, it almost looks like it is rendered deliberately. The patterns are also visible in all shadow edges. Like some kind of dithering. 
Ok, checked it with wined3d. The shadow edges are indeed dithered. But where wined3d has it on a pixel scale, d9vk renders much larger patterns. The bar is not there in wined3d though.

**Comment by edwin-v** [comment]
Even though the project has now been integrated with the main dxvk, some interesting information for archival purposes.

I tried this recently and the issue was not visible. I assumed this had been fixed in some recent version of DXVK. However, at some point I removed the launcher, which works in proton these days, by replacing the exe and suddenly the issue came back. Restoring the launcher made it correct again. So in the end this problem was not an actual rendering issue, but some sort of initialisation issue caused by skipping the launcher or launching BL2 with a different exe name.
